src/ahp/indexer.rs

```diff
@@ -20,8 +20,10 @@ use std::marker::PhantomData;
 #[derivative(Clone(bound = ""), Copy(bound = ""), Default(bound = ""), Debug(bound = ""), Eq(bound = ""), PartialEq(bound = ""))]
 #[derive(CanonicalSerialize, CanonicalDeserialize)]
 pub struct IndexInfo<F> {
-    /// The total number of variables in the constraint system.
-    pub num_variables: usize,
+    /// The total number of witnesses in the constraint system.
+    pub num_witness: usize,
```

Comment:
maybe worth to mention: the number of witnesses includes constants.

Reply:
aaah, sorry. Shouldn't be the constants to be treated as public inputs, always enforced to be equal to 1 in the verifier?

Reply:
I think constants are treated as witnesses, and enforced to be equal to the 1 "input" of the CS